src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -285,6 +285,10 @@ public function request(string $method, string $url, array $options = []): Respo
             $curlopts[\CURLOPT_TIMEOUT_MS] = 1000 * $options['max_duration'];
         }
 
+        if (0 < $options['max_connect_duration']) {
+            $curlopts[\CURLOPT_CONNECTTIMEOUT_MS] = (int) (1000 * $options['max_connect_duration']);
```

Comment:
```suggestion
            $curlopts[\CURLOPT_CONNECTTIMEOUT_MS] = ceil(1000 * $options['max_connect_duration']);
```

Reply:
Coming back to this one, why should we add `ceil()` here? It would be inconsistent with `max_duration`, isn't it?